crates/gen/src/com_interface.rs

```diff
@@ -28,21 +30,43 @@ impl ComInterface {
     pub fn from_type_name(name: TypeName) -> Self {
         let mut count = BTreeMap::new();
 
-        let methods = name
-            .def
-            .methods()
-            .map(|method| {
+        let mut bases = Vec::new();
+        let mut next = name.def;
+
+        loop {
+            let base = next.interfaces().next().unwrap().interface().resolve();
+
+            if base.name() == ("Windows.Win32.Com", "IUnknown") {
+                break;
+            }
+
+            bases.insert(0, TypeName::new(&base, Vec::new(), name.namespace));
+            next = bases[0].def;
+        }
+
+        let mut methods = Vec::new();
+
+        for def in bases
+            .iter()
+            .map(|name| name.def)
+            .chain(std::iter::once(name.def))
+        {
+            for method in def.methods() {
```

Comment:
Nit: you can get rid of the inner `for` loop by adding `flat_map(|def| def.methods())` after the call to `chain`.

Reply:
Problem is I get this:

```
error[E0515]: cannot return value referencing function parameter `def`
  --> crates\gen\src\com_interface.rs:53:28
   |
53 |             .flat_map(|def|def.methods())
   |                            ---^^^^^^^^^^
   |                            |
   |                            returns a value referencing data owned by the current function
   |                            `def` is borrowed here
```